Source/FocusTuner/FocusTuner.cs

```diff
@@ -0,0 +1,138 @@
+                commonConfiguration.LoadTermination[0] = new Complex { Real = 0, Imaginary = 0 };
+
+                return commonConfiguration;
+            }
+        }
+        #endregion
+
+        #region Configuration
+        /// <summary>Reinitializes all the axes of the tuner to its home position before terminating the software connection to the instrument.</summary>
+        /// <param name="iTuner">Specifies a reference to the instrument.</param>
+        public static void CloseTuner(FocusITunerBroker iTuner)
+        {
+            iTuner.InitializeTunerAllAxises();
+            iTuner.Close();
+        }
+
+        /// <summary>Configures common settings for load pull using the Focus tuner.</summary>
+        /// <param name="iTuner">Specifies a reference to the instrument.</param>
+        /// <param name="commonConfiguration">Specifies the settings for load pull using the Focus tuner.</param>
+        /// <returns>The primary frequency in GHz, of the active tuner calibration data set.</returns>
+        public static double ConfigureCommon(FocusITunerBroker iTuner, CommonConfiguration commonConfiguration)
+        {
+            List<Complex[]> sParameters = new List<Complex[]>();
+            foreach (var element in commonConfiguration.DUTtoTunerSParameters)
+            {
+                Complex[] sParameter = new Complex[4];
+                //The order in the Focus driver is S11, S12, S21 and S22.
+                sParameter[0] = element.S11;
+                sParameter[1] = element.S12;
+                sParameter[2] = element.S21;
+                sParameter[3] = element.S22;
+                sParameters.Add(sParameter);
+            }
+            iTuner.ConfigureTunerMode(commonConfiguration.TunerMode);
+            iTuner.ConfigureActiveCalibration(commonConfiguration.CalibrationID);
+            iTuner.ConfigureAdapter(sParameters.ToArray());
+            iTuner.ConfigureTermination(commonConfiguration.LoadTermination);
+            double[] frequencies = iTuner.QueryActiveFrequency();
+            return frequencies[0];
+        }
+
+        /// <summary>Sets the specified reflection coefficient in the device reference plane at fundamental frequency with timeout.</summary>
+        /// <param name="iTuner">Specifies a reference to the instrument.</param>
+        /// <param name="gamma">Specifies the desired reflection coefficient in the device reference plane. Valid Range is Real part: [-1, 1] and Imaginary part: [-1, 1].</param>
+        /// <param name="timeout">specifies the maximum length of time, in seconds, to allow the operation to complete. The default is 60.</param>
+        /// <returns>returns the estimated reflection coefficient at the device reference plane for the current tuner position at all frequencies of the active tuner calibration data set.</returns>
+        public static Complex[] MoveTunerPerGamma(FocusITunerBroker iTuner, Complex gamma, short timeout = 60)
```

Comment:
I'm slightly bothered that focus chose to use a short for the timeout. I see that's the same in LV as well. I guess we can keep if for now, but it still makes me feel uneasy ;P an int or a double would have been a more natural selection.

Reply:
I agree. For now, let's follow the type in iTunerX interface.

        [DispId(19)]
        public virtual short OPC(short maxWaitSec);